packages/trader/src/AppV2/Containers/Trade/trade-types.tsx

```diff
@@ -125,6 +126,24 @@ const TradeTypes = ({ contract_type, onTradeTypeSelect, trade_types }: TTradeTyp
         }));
     };
 
+    const scrollToSelectedTradeType = () => {
+        setTimeout(() => {
+            let position_x = 0;
+            if (trade_types_ref?.current) {
+                const selected_chip = trade_types_ref?.current.querySelector(
```

Comment:
Will it work, if we pass ref not to the container, but to the selected chip directly, something like this?
```
            {saved_pinned_trade_types.length > 0 &&
                saved_pinned_trade_types[0].items.map(({ title, id }: TItem) => (
                    <Chip.Selectable key={id} onChipSelect={onTradeTypeSelect} selected={isTradeTypeSelected(id)}>
                        <Text size='sm' ref={isTradeTypeSelected(id) ? your_ref : null}>{title}</Text>
                    </Chip.Selectable>
                ))}
```
                
In that case we won't need to search for selected chip with the help of `querySelector`

Reply:
Seems, that Chip from Quill can accept ref

Reply:
the scroll that I have to do is with respect to container. I had thought of the way you said to pass ref to Chip or to Text component but in that case I have to get the parentNode and then scroll but in case of any future development , if someone adds a element in div in between chip and container the code will break. So I found the approach that I took to be okay. But do let me know in case if there is a better way.

Reply:
Btw, will `scrollIntoView` work? Just curious, because it might be called from selected label ref, without any container ref. 
`selected_chip_ref.current.scrollIntoView({ behavior: 'smooth', inline: 'start' });`

Reply:
scrollIntoView can be made to work , if ref is an array and once a chip is selected we check the ref which has attribute of data-selected is true and then we scrollIntoView. With one ref and passing `ref={isTradeTypeSelected(id) ? your_ref : null}` is not working. But let's leave it with ScrollBy because it give fine control over scroll in horizontal direction.